### PR TITLE
Re-enable checker ``bad-docstring-quotes`` for Python <= 3.7.

### LINKED ISSUE #6087: Re-enable ``bad-docstring-quotes`` for Python 3.7
### Current problem

With changes to `doc_node` in astroid we can now check docstring quotes in `3.7` as well. This was turned off in #5526.

### Desired solution

Turn the checker on again.

### Additional context

@kasium Do you still want to do this? Should I assign you?

### PR DESCRIPTION
**Pull request by jpy-git**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #6087

Re-enable checker ``bad-docstring-quotes`` for Python <= 3.7.

### CODE REVIEW

File: doc/whatsnew/2.13.rst [hunk 1]
```diff
@@ -120,6 +120,10 @@ Extensions
 
 * ``DocStringStyleChecker``
 
+    * Re-enable checker ``bad-docstring-quotes`` for Python <= 3.7.
+
+      Closes #6087
```
Comment (jpy-git) [inline doc/whatsnew/2.13.rst:125]: We could maybe remove this and the note about removing it below from this whatsnew given it's in the same minor version?

Comment (DanielNoord) [reply doc/whatsnew/2.13.rst:125]: Makes sense!

**Review by Pierre-Sassoulas — APPROVED** [review]
👌

File: doc/whatsnew/2.13.rst (hunk 1 above)
Comment (jpy-git) [reply doc/whatsnew/2.13.rst]: I noticed we assigned this PR to the 2.14 milestone so I've moved this line to 2.14 instead 😄

**Pull request merged**